packages/itwinui-react/package.json

```diff
@@ -54,28 +58,29 @@
     "ux"
   ],
   "scripts": {
-    "build": "yarn clean:build && tsc -p tsconfig.cjs.json && tsc -p tsconfig.esm.json && yarn build:styles && yarn build:post",
+    "build": "yarn clean:build && tsc -p tsconfig.cjs.json && tsc -p tsconfig.esm.json && tsc -p tsconfig.react-table.json && yarn build:styles && yarn build:post",
     "build:styles": "vite build src/styles.js",
     "build:watch": "concurrently \"tsc -p tsconfig.cjs.json --watch\" \"tsc -p tsconfig.esm.json --watch\"",
     "build:post": "node ./scripts/postBuild.mjs",
-    "clean:build": "rimraf esm && rimraf cjs",
+    "clean:build": "rimraf esm && rimraf cjs && rimraf react-table.d.ts",
     "clean:coverage": "rimraf coverage",
     "clean": "rimraf .turbo && yarn clean:coverage && yarn clean:build && rimraf node_modules",
-    "test": "jest",
-    "test:watch": "jest --watch",
+    "test": "yarn test:types && yarn test:unit",
+    "test:unit": "jest",
```

Comment:
I could not find the thread about testing.
Why it was decided to move type test with unit tests?

Reply:
These are some threads I could find that relate to testing:
* https://github.com/iTwin/iTwinUI/pull/1514#discussion_r1312162204
* https://github.com/iTwin/iTwinUI/pull/1514#discussion_r1321811752

To summarize the discussion in the threads related to testing, we decided for now to remove `test:types` from `storybook/package.json`. Instead we added `tsc` step to the `build` script so that type checking for all files can happen in `build`.

But for `itwinui-react/package.json`, I proposed to keep `test:types` since it would be good to type check Table unit test files too. In a future PR, we can type check all files in `itwinui-react` (even those files line unit tests that are not part of `build`) for all components apart from Table too. By this I mean, `tsc -p tsconfig.test.json --noEmit` where the tsconfig points to all files in `itwinui-react`.